src/backend.py

```diff
@@ -0,0 +1,757 @@
+            players[str(i)] = {"initials": initials, "name": full_name}             
+    return json_dumps(players), 200
+
+    
+@add_route("/api/player/update", method="POST", auth=True)
+def app_updatePlayer(request):    
+    #TODO this correctly deletes the player and unlinks their scores, but should it remove their name from leaderbaord scores? (currently it does not)
+    body = request.data
+    
+    index = int(body['id'])  
+    if index < 0 or index > ds_memory_map["names"]["count"]:
+        raise ValueError(f"Invalid index: {index}")
+
+    initials = body['initials'].upper()[:3]
+    name = body['full_name'][:16]
+
+    # if name and initials are empty, delete the record
+    if len(initials) == 0 and len(name) == 0:
+        print("Deleting record")
+        from SPI_DataStore import blankIndPlayerScores
+        blankIndPlayerScores(int(body['id']))
+    
+    print(f"Updating record {index} with {initials} and {name}")
+
+    ds_write_record("names",{"initials":initials,"full_name":name},index)
+
+
+@add_route("/api/player/scores")
+def app_getScores(request):
+    player_id = int(request.query.get("id"))
+    scores = []                      
+    numberOfScores = ds_memory_map["individual"]["count"]
+    for i in range(numberOfScores):
+        record = ds_read_record("individual", i, player_id)  
+        score = record['score']
+        date = record['date'].strip().replace('\x00', ' ')          
+        if score > 0:
+            scores.append({
+                "score": score,
+                "date": date
+            })                           
+    return json_dumps(scores), 200
+
+
+@add_route("/api/player/scores/reset", auth=True)
+def app_resetIndScores(request):
+    from SPI_DataStore import blankIndPlayerScores
+    index = int(request.args.get("id"))
+    blankIndPlayerScores(index)
+
+
+#
+# Settings
+#
+@add_route("/api/settings/score_claim_methods")
+def app_getScoreCap(request):
+    raw_data = ds_read_record("extras", 0)["other"]
+    print(f"raw_data: {raw_data}")
+    score_cap = bool(raw_data)
+    return json_dumps({"on-machine": score_cap}), 200
+
+
+@add_route("/api/settings/score_claim_methods", method="POST", auth=True)
+def app_setScoreCap(request):
+    json_data = request.data
+    if "on-machine" in json_data:        
+        info = ds_read_record("extras", 0)
+        info["other"] = json_data['on-machine']
+        ds_write_record("extras", info, 0)
+
+
+
+@add_route("/api/settings/tournament_mode", method="POST", auth=True)
+def app_setTournamentMode(request):    
+    import SharedState
+    SharedState.tournamentModeOn = int(request.json['tournament_mode'])
+
+
+@add_route("/api/settings/factory_reset", auth=True)
+def app_factoryReset(request):
+    # TODO confirm with this is an ok way to reset the device
+    from SPI_DataStore import blankAll
+    from machine import reset
+    import reset_control
+
+    reset_control.reset() # turn off pinbal machine
+    blankAll()
+    reset()
```

Comment:
Is this an ok method for factory reset?

Reply:
Does reset - reset the PICO?    I think this is fine.  I'll need to review blankAll() - was only a utility to start with.  I'll check it for production use / error handling

Reply:
it's hard for me to confirm how *hard* of a reboot it is, but it for sure starts main again and comes back up in AP mode. machine.reset() is supposed to be a full/hard reboot.

https://docs.micropython.org/en/latest/library/machine.html#machine.reset